### ISSUE TITLE
Doesn't work on Openshift

### ISSUE THREAD

**Issue by vrutkovs** [zalando-incubator/postgres-operator#359]
Postgres-operator requires endpoints to be modified, but setting endpoint address to pod IP is forbidden in Openshift (3.10 if matters). The pod would complain about it with:
```
HTTP response headers: HTTPHeaderDict({'Audit-Id': 'a04ade0f-8f0e-4cae-94d6-77ad01ef0e5c', 'Cache-Control': 'no-store', 'Content-Type': 'application/json', 'Date': 'Sat, 04 Aug 2018 14:17:21 GMT', 'Content-Length': '267'})
```

Is there a way to avoid endpoints usage? IIRC Patroni can sync over configmaps, but it appears the operator doesn't have a setting to use it

**Comment by vrutkovs** [comment]
Thanks, this seems to do something! Some more changes are needed:
* correct env name for spilo is currently `KUBERNETES_USE_CONFIGMAPS`
* additional permission to create ConfigMaps for spilo SA is required
* additional service needs to be created to expose masters

However it seems replicas are broken, as these complain about system ID mismatch

**Comment by dejwsz** [comment]
I tried the thing with "pod_environment_configmap" and setting PATRONI_KUBERNETES_USE_ENDPOINTS to false. No luck with this in Openshift 3.11.

**Comment by flickerfly** [comment]
Check out @vrutkovs follow-up comment for the correct variable.

**Comment by dejwsz** [comment]
Indeed. But spilo-role is not assigned so "pod_role_label" is not working. Any trick here?

**Comment by dejwsz** [comment]
I saw such warning in log:

2020-03-03 10:18:45,606 - bootstrapping - WARNING - could not parse kubernetes labels as a JSON: Expecting value: line 1 column 1 (char 0), reverting to the default: {"application": "spilo"}

and two CRIT

2020-03-03 10:18:46,711 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
2020-03-03 10:18:46,721 CRIT Server 'unix_http_server' running without any HTTP authentication checking

but in general, it looks like it's starting ok but it does not label pods as it should - "spilo-role" is not assigned at all.

**Comment by stgrace** [comment]
Ran into this issue using postgres operator version v1.5.0. This issue does not seem to be fixed. Why was it closed?
